src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java

```diff
@@ -122,6 +122,31 @@ public void addUpdateRecordToIndex(String indexName, UpdateRecord updateRecord)
         riverDAO.save(river);
     }
 
+    @Transactional(readOnly = true)
+    public List<Map<String, Object>> getAllConfigs() {
+        return riverDAO.findAll().stream().map(River::toMap).collect(Collectors.toList());
+    }
+
+    @Transactional()
+    public List<Integer> setAllConfigs(List<String> configs) {
+        int newConfigs = 0;
+        int updatedConfigs = 0;
+        for (String config : configs) {
+            River newRiver = createRiver(config);
+            River foundRiver = riverDAO.findById(newRiver.getRiverName()).orElse(null);
```

Comment:
Not clear to me what is the interplay between id and rivername ...

Reply:
riverName is the ID of obj River. Should i change it to ID?